codersdk/deployment.go

```diff
@@ -2287,3 +2305,13 @@ func (c *Client) SSHConfiguration(ctx context.Context) (SSHConfigResponse, error
 	var sshConfig SSHConfigResponse
 	return sshConfig, json.NewDecoder(res.Body).Decode(&sshConfig)
 }
+
+func defaultUpgradeMessage() string {
+	// Our installation script doesn't work on Windows, so instead we direct the user
+	// to the GitHub release page to download the latest installer.
+	version := buildinfo.Version()
+	if runtime.GOOS == "windows" {
+		return fmt.Sprintf("download the server version from: https://github.com/coder/coder/releases/v%s", version)
+	}
+	return fmt.Sprintf("download the server version with: 'curl -L https://coder.com/install.sh | sh -s -- --version %s'", version)
+}
```

Comment:
I think my comment got removed, reposting it (might see this twice??)

This is now running on the server, not the client. So the OS is likely to never be windows, and windows clients will get the linux/mac msg.

I think having the client send it's OS is "eh", so maybe just make the default an empty string. On the client side, if the server sends over an empty string, then use this function. Pushing the caller of this back to the client.

Reply:
:facepalm: